spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -90,6 +98,32 @@ class CometSparkSessionExtensions
               scanExec.copy(scan = cometScan),
               runtimeFilters = scanExec.runtimeFilters)
 
+          // unsupported parquet data source V2
+          case scanExec: BatchScanExec if scanExec.scan.isInstanceOf[ParquetScan] =>
+            val requiredSchema = scanExec.scan.asInstanceOf[ParquetScan].readDataSchema
+            val info1 = if (isSchemaSupported(requiredSchema)) {
+              CometExplainInfo(s"Schema $requiredSchema is not supported")
+            } else {
+              CometExplainInfo.none
+            }
```

Comment:
We could use Scala's `Option` type here rather than defining our own `none` constant. For example:

```scala
            val info1 = if (isSchemaSupported(requiredSchema)) {
              Some(CometExplainInfo(s"Schema $requiredSchema is not supported"))
            } else {
              None
            }
```

We would also need to update the call to pass the list of reasons to `opWithInfo` to use `flatten` so that we only pass the valid reasons.

```scala
opWithInfo(scanExec, CometExplainInfo("SCAN", Seq(info1, info2, info3).flatten))
```

Reply:
Good idea. I may have a cleaner way of doing this thoguh, which would make this unnecessary. Give me a little time since I need to change a lot of places to try out the idea. If that doesn't work, I'll change this to use Scala `Option`